R/source.R

```diff
@@ -149,24 +155,10 @@ rust_source <- function(file, code = NULL,
   # Get target name, not null for Windows
   specific_target <- get_specific_target_name()
 
-  status <- system2(
-    command = "cargo",
-    args = c(
-      sprintf("+%s", toolchain),
-      "build",
-      "--lib",
-      if (!is.null(specific_target)) sprintf("--target %s", specific_target) else NULL,
-      sprintf("--manifest-path %s", file.path(dir, "Cargo.toml")),
-      sprintf("--target-dir %s", file.path(dir, "target")),
-      if (profile == "release") "--release" else NULL
-    ),
-    stdout = stdout,
-    stderr = stdout
+  invoke_cargo(
+    toolchain, specific_target, dir,
+    profile, stdout, stderr, use_rtools
```

Comment:
When there are so many arguments, it's generally better to name them explicitly to avoid difficult-to-debug problems with argument order.